### ISSUE TITLE
How to use .OnError(...) with json

### ISSUE THREAD

**Issue by lalasaala** [danielgerlag/workflow-core#88]
Hi Daniel,
  I do not see in the documentation how to migrate .OnError(...) to json.

**Comment by danielgerlag** [comment]
Try this

```
{
  "Id": "HelloWorld",
  "Version": 1,
  "Description": "",
  "DataType": "ScratchPad.MyDataClass, ScratchPad",
  "Steps": [
    {
      "Id": "Hello",
      "StepType": "ScratchPad.HelloWorld, ScratchPad",
      "NextStepId": "Generate",
      "ErrorBehavior": "Suspend"
    },
...
```

**Comment by lalasaala** [comment]
Regarding the closed referenced Issue #89
I am having difficulty translating the event key to json and a few more in the below example. 
Can you please translate this whole workflow example to json? This is from Sample 04
builder
.StartWith(context => ExecutionResult.Next())
.WaitFor("MyEvent", (data, context) => context.Workflow.Id, data => DateTime.Now)
.Output(data => data.StrValue, step => step.EventData)
.Then()
.Input(step => step.Message, data => "The data from the event is " + data.StrValue)
.Then(context => Console.WriteLine("workflow complete"));

**Comment by danielgerlag** [comment]
Hi,

You would need to implement the step bodies are compiled C# classes as we don't support inline steps via json, but that flow would look something like this:


```json
{
  "Id": "HelloWorld",
  "Version": 1,
  "DataType": "MyApplication.MyDataClass, MyApplication",
  "Steps": [
    {
      "Id": "Hello",
      "StepType": "MyApplication.HelloWorld, MyApplication",
      "NextStepId": "Wait"
    },
    {
      "Id": "Wait",
      "StepType": "WorkflowCore.Primitives.WaitFor, WorkflowCore",
      "NextStepId": "Print",
      "Inputs": {
         "EventName": "\"MyEvent\"",
         "EventKey": "context.Workflow.Id",
         "EffectiveDate": "DateTime.Now"
       },
      "Outputs": {
         "StrValue": "step.EventData"
      }
    },
   {
      "Id": "Print",
      "NextStepId": "Bye",
      "StepType": "MyApplication.PrintMessage, MyApplication",
      "Inputs": {
         "Message": "data.StrValue"
       }
    },    
    {
      "Id": "Bye",
      "StepType": "MyApplication.GoodbyeWorld, MyApplication"
    }
  ]
}
```
